fortigate-autoscale/azure/azure-platform-adapter.ts

```diff
@@ -731,18 +778,53 @@ export class AzurePlatformAdapter implements PlatformAdapter {
             nextHeartBeatTime: rec.nextHeartbeatTime,
             syncState: syncStateString,
             syncRecoveryCount: rec.syncRecoveryCount,
-            seq: rec.seq
+            seq: rec.seq,
+            sendTime: rec.sendTime,
+            deviceSyncTime: rec.deviceSyncTime,
+            deviceSyncFailTime: rec.deviceSyncFailTime,
+            // store boolean | null
+            deviceSyncStatus:
+                (rec.deviceSyncStatus === null && 'null') ||
+                (rec.deviceSyncStatus && 'true') ||
+                'false',
+            // store boolean | null
+            deviceIsPrimary:
+                (rec.deviceIsPrimary === null && 'null') ||
+                (rec.deviceIsPrimary && 'true') ||
+                'false',
+            deviceChecksum: rec.deviceChecksum
         });
-        const check: ConsistenyCheckType<typeof item> = {
-            vmId: rec.vmId,
-            scalingGroupName: rec.scalingGroupName,
-            ip: rec.ip
+
+        const checker = (
+            snapshot: typeof item
+        ): Promise<{ result: boolean; errorMessage: string }> => {
+            let errorMessage = this.dataConsistencyCheck(
+                {
+                    vmId: rec.vmId,
+                    scalingGroupName: rec.vmId,
+                    ip: rec.vmId
+                },
+                snapshot
+            );
+            // NOTE: strictly update the record when the sequence to update is equal to or greater
```

Comment:
keeping multiple heartbeat records instead of overwriting them might improve this situation.

Reply:
good idea! The current heartbeat monitoring has been improved. Will take this idea into consideration next time.